### PR TITLE
Added controller for save-user & save-game

### PR DESCRIPTION
**Pull request by 318097**

### CODE REVIEW

File: client/src/api/socketService.js [hunk 1]
```diff
@@ -22,9 +22,19 @@ const leaveGame = () => {
   socket.emit('disconnect');
 };
 
+// const updateGameState = ({ pacman, roomId, playerId }) => {
```
Comment (kaushikb11) [inline client/src/api/socketService.js]: Why comments are here?

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 2]
```diff
@@ -131,26 +134,12 @@ class PacmanGame extends Component {
   }
 
   startGame = () => {
-    const { config, pacman, status } = this.state;
-    if (status === 1) {
```
Comment (kaushikb11) [inline client/src/components/PacmanGame/PacmanGame.jsx]: Why did you remove this?

File: server/api/socketService.js [hunk 3]
```diff
@@ -1,41 +1,61 @@
...
-
-  socket.emit('connected');
+  // socket.on('update-game-state', ({ pacman, playerId, roomId }) => {
+  //   const room = games[roomId];
+  //   console.log(room.pacmanOne.playerId);
+  //   const player = room.pacmanOne.playerId === playerId ? 'pacmanOne' : 'pacmanTwo';
+  //   room[player] = pacman;
+  //   games[roomId] = room;
+  //   socket.emit('game-update', { pacmanOne: room.pacmanOne, pacmanTwo: room.pacmanTwo });
+  // });
```
Comment (shivangg) [inline server/api/socketService.js]: This part for getting the gameState. Why commented?

**Review by shivangg — APPROVED** [review]

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 4]
```diff
@@ -131,26 +134,12 @@ class PacmanGame extends Component {
...
-      let { score } = this.state;
-      clearInterval(this.animationHandler);
-      score -= 10;
-      this.setState({ status: 0, score });
-      return;
-    }
-    if (status === 2) {
-      this.setInitialGameState();
-      this.setState({
```
Comment (kaushikb11) [inline client/src/components/PacmanGame/PacmanGame.jsx]: Don't just remove it. We need it for the singleplayer

**Review by kaushikb11 — DISMISSED** [review]

**Review by Shalinit3 — APPROVED** [review]

File: client/src/components/PacmanGame/PacmanGame.jsx (hunk 4 above)
Comment (318097) [reply client/src/components/PacmanGame/PacmanGame.jsx]: So put it in a conditional. When I kept changes currentGameState() wasnt found.
